snippet/util_test.go

```diff
@@ -0,0 +1,54 @@
+package snippet
+
+import (
+	"testing"
+
+	"github.com/go-test/deep"
+)
+
+func TestTomlRegEx(t *testing.T) {
+	tests := []struct {
+		name string
+		path string
+		want bool
+	}{
+		{name: "match - filename", path: "snippet.toml", want: true},
+		{name: "match - absolute path", path: "/home/username/.config/pet/config.toml", want: true},
+		{name: "match - absolute path with home alias", path: "~/.config/pet/snippet2.toml", want: true},
+		{name: "match - relative path", path: "../../some/directory/best.toml", want: true},
+		{name: "mismatch - filename", path: "file.yaml", want: false},
+		{name: "mismatch - absolute path", path: "/home/username/.config/pet/config.json", want: false},
+		{name: "mismatch - absolute path with home alias", path: "~/.config/pet/snippet2.xml", want: false},
+		{name: "mismatch - relative path", path: "../../some/directory/unrelated.html", want: false},
+		{name: "mismatch - extension with dot", path: ".toml", want: false},
+		{name: "mismatch - extension without dot", path: "toml", want: false},
+		{name: "mismatch - similar extension prefix", path: "file.tomlx", want: false},
+		{name: "mismatch - similar extension suffix", path: "file.xtoml", want: false},
+		{name: "mismatch - similar extension infix", path: "file.xtomlx", want: false},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tomlRegEx.MatchString(tt.path)
+
+			if got != tt.want {
+				t.Errorf("Expected result %v, but got %v", tt.want, got)
+			}
+		})
+	}
+}
+
+func TestGetFiles(t *testing.T) {
+	t.Run("success - returns list of toml files", func(t *testing.T) {
+		got := getFiles("testdata")
+		want := []string{
+			"testdata/01-snippet.toml",
```

Comment:
For these kinds of tests where files / folders are necessary, it would be best to create the things needed and then delete them at the end of the tests. (setup/cleanup)

You can imagine how unwieldy it becomes when many empty files are added for the sake of tests.

Take a look at this for example: https://medium.com/nerd-for-tech/setup-and-teardown-unit-test-in-go-bd6fa1b785cd

A setup + defer delete at the beginning of a test would be nicer.

Reply:
I will take a look. I might actually mock the walk function, it's better to only test the `pet` code rather than actually reading the filesystem.

Reply:
Yeah it's not an easy balance - when to start/stop mocking stuff. In an ideal world you mock everything you're not testing, but you also get some benefits from 'integration' testing things + sometimes the code without mocking is alot simpler + more readable which makes it more valuable.

You don't want to write too much code / complicate the tests. If I were you I would see if just creating the files dynamically is easier/more readable than mocking the walk, or what the upsides/downsides of mocking the Walk would be and then decide.

Reply:
Creating the files on the fly is much easier and doesn't affect the main codebase so I went with it. I did not create separate setup/teardown functions. Since I already use `t.Run` it looked readable to me but need another pair of eyes to confirm that.

I also did not assert file creation errors to keep it simple but not confident about this choice. So comment on that too.

Last thing, the last commit is separate on purpose, if you think it's unnecessary - which it is for now - you can ignore it.